### PR TITLE
Refactor random ports assignation in tribler config

### LINKED ISSUE #6148: OSError asyncio.base_events in create_server: [Errno 10048] error while attempting to bind on address ('127.0.0.1', 8307)
Sentry Issue: [TRIBLER-16](https://sentry.tribler.org/organizations/tribler/issues/316/?referrer=github_integration)

```
OSError: [Errno 10048] error while attempting to bind on address ('127.0.0.1', 8307): обычно разрешается только одно использование адреса сокета (протокол/сетевой адрес/порт)
  File "lib\asyncio\events.py", line 81, in _run
  File "ipv8\taskmanager.py", line 131, in done_cb
  File "src\tribler-core\tribler_core\modules\tunnel\socks5\server.py", line 27, in start
  File "lib\asyncio\base_events.py", line 1463, in create_server
```

### PR DESCRIPTION
**Pull request by drew2a**
This PR removes random port assignation from Tribler config and places it to the corresponding modules. It also reduces the time between port checking and port using.

Fixes #6148

### CODE REVIEW

File: src/tribler-core/tribler_core/modules/libtorrent/download_manager.py [hunk 1]
```diff
@@ -167,12 +166,15 @@ def is_shutdown_ready(self):
...
         # Due to a bug in Libtorrent 0.16.18, the outgoing_port and num_outgoing_ports value should be set in
         # the settings dictionary
+        self._logger.info('Creating a session')
         settings = {'outgoing_port': 0,
                     'num_outgoing_ports': 1,
                     'allow_multiple_connections_per_ip': 0}
 
         # Copy construct so we don't modify the default list
         extensions = list(DEFAULT_LT_EXTENSIONS)
+        anon_port = self.tribler_session.config.get_anon_listen_port() or NetworkUtils().get_random_free_port()
```
Comment (ichorid) [inline src/tribler-core/tribler_core/modules/libtorrent/download_manager.py:176]: Bravo! :clap:

**Comment by drew2a** [comment]
retest this please

**Comment by drew2a** [comment]
retest this please

**Comment by drew2a** [comment]
retest this please

**Comment by drew2a** [comment]
retest this please

**Review by ichorid — APPROVED** [review]

**Pull request merged**